kubernetes/kubernetes.go

```diff
@@ -569,6 +567,20 @@ func (ka KubeAgentConfig) sendMetricsToCustomBlob(ctx context.Context, metricSam
 	ka.uploadBlob(ctx, azureClient, metricSample)
 }
 
+func (ka KubeAgentConfig) sendMetricsToCustomBlobManagedIdentity(ctx context.Context, metricSample *os.File) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		log.Fatalf("Could not establish Azure credentials with managed identity, "+
```

Comment:
Can we add these two as a startup check for azure as well? If there's an issue it would take 10m for it to fail.